packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -106,44 +195,39 @@ export class VaDate {
       error,
       handleDateBlur,
       handleDateChange,
+      handleDateKey,
       value,
     } = this;
 
-    // Convert string to number to remove leading 0 on values less than 10
-    const [year, month, day] = (value || '')
-      .split('-')
-      .map(val => parseInt(val, 10));
-
-    const daysForSelectedMonth = month > 0 ? days[month] : [];
-
-    // Check validity of date if invalid provide message and error state styling
-    const dateInvalid =
-      required && (!isFullDate(value) || day > daysForSelectedMonth.length)
-        ? 'Please provide a valid date'
-        : null;
+    const [year, month, day] = (value || '').split('-').map(val => val);
+    const monthNum = parseInt(month);
+    const dayNum = parseInt(day);
+    const daysForSelectedMonth = monthNum > 0 ? days[monthNum] : [];
 
-    // Setting new attribute to avoid conflicts with only using error attribute
     // Error attribute should be leveraged for custom error messaging
     // Fieldset has an implicit aria role of group
     return (
-      <Host value={value} invalid={dateInvalid}>
-        <fieldset aria-label="Select Month and two digit day XX and four digit year format XXXX">
+      <Host value={value} error={error} onBlur={handleDateBlur}>
+        <fieldset aria-label="Select month and day fields are in two digit format XX and input year field is in four digit format XXXX">
           <legend>
-            {label} {required && <span class="required">(*Required)</span>}
+            {label}{' '}
+            {required && (
+              <span class="required">(*{i18next.t('required')})</span>
```

Comment:
@k80bowman the only other hardcoded values are "Month", "Day", and "Year" do we not have translations for those?

Reply:
@k80bowman we have the translations for Month, Day, and Year as well as the individual months so I applied them via this PR. All hardcoded text is now covered.

Reply:
What about `'Please enter a valid date'` from line 118 and `"Please enter two digits for the month"` from line 221 and `"Please enter two digits for the day"` from line 238 and `"Please enter four digits for the year"` from line 260?

Reply:
@k80bowman "Please enter a valid date" is a specific case because we decided to bake in validation to these 2 components. I'm open to whatever but @bkjohnson had originally suggested putting i18next in place. I'll do whatever you all would like. Just let me know if I should leave it in or rip it out for both components.

Reply:
Per standup today i18next has been remove from `va-date` and `va-date-text-input`